docs/source/examples/semantic_segmentation.rst

```diff
@@ -0,0 +1,223 @@
+
+.. code-block:: python
+
+    input_transform = tfms.Compose([
+        tfms.Resize((image_size, image_size)),  
+        tfms.ToTensor(),        
+        tfms.Normalize(imagenet_mean, imagenet_std)
+    ])
+    
+    target_transform = tfms.Compose([
+        tfms.Resize((image_size, image_size)),  
+        tfms.ToTensor(),        
+    ])
+    
+    # Creating the dataset
+    train_dataset = datasets.VOCSegmentation('./dataset/', year='2007', download=True, image_set='train', transform=input_transform, target_transform= target_transform)
+    valid_dataset = datasets.VOCSegmentation('./dataset/', year='2007', download=True, image_set='val', transform=input_transform, target_transform= target_transform )
+    
+    # Creating the dataloader
+    train_dataloader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+    valid_dataloader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)
+
+A random batch of the VODSegmentation dataset images
+====================================================
+
+Let's see some of the input samples inside the training dataset.
+
+.. code-block:: python
+
+    samples = next(iter(valid_dataloader))
+    inputs = samples[0]
+    input_grid = make_grid(inputs)
+    
+    fig = plt.figure(figsize=(10, 10))
+    input_grid = input_grid.numpy()
+    input_grid = input_grid.transpose((1, 2, 0)) * imagenet_std + imagenet_mean
+    inp = np.clip(input_grid, 0, 1)
+    plt.imshow(inp)
+    plt.axis('off')
+    plt.show()
+    
+.. image:: /_static/img/semantic_segmentation/voc_segment_batch.png 
+
+The ground-truth (segmentation map) for the image grid shown above is as below.
+
+.. code-block:: python
+
+    ground_truth = samples[1]
+    input_grid = make_grid(ground_truth)
+    inp = input_grid.numpy()[0]
+    
+    fig = plt.figure(figsize=(10, 10))
+    plt.imshow(inp)
+    plt.axis('off')
+    plt.show()
+    
+.. image:: /_static/img/semantic_segmentation/voc_segment_batch_gt.png 
+
+It is worth mentioning that, as we have approached the segmentation task as an image translation problem, we take advantage of MSELoss for the training. Moreover, we believe that using the U-Net with a pre-trained encoder would help the network converge sooner and better. As this convolutional encoder is previously trained on the ImageNet, it is able to recognize low-level features (such as edge, color, etc.) and high-level features at its beginning and final layers respectively.
+
+.. code-block:: python
+
+    # specifying loss function
+    criterion = nn.MSELoss()
+    
+    # specifying the network
+    network = smp.Unet('resnet34', encoder_weights='imagenet')
+    
+    # specifying optimizer
+    optimizer = optim.Adam (network.parameters(), lr=learning_rate)  
+
```

Comment:
```suggestion

We can see the architecture of the ResNet-34-U-Net below. As noticed in the section above, this network is imported from the segmentation-models-pytorch library which contains many other architectures as well. You can import and use other available networks to try to increase the accuracy.
```
Can you put this sentence back since I didn't want you to remove it?